filters/private/NormalResult.hpp

```diff
@@ -0,0 +1,80 @@
+#include <Eigen/Dense>
+
+#include <pdal/KDIndex.hpp>
+
+namespace pdal
+{
+
+using namespace Eigen;
+
+struct NormalResult
+{
+    struct error : std::runtime_error
+    {
+        error(const std::string& err) : std::runtime_error(err)
+        {}
+    };
+
+    NormalResult() : m_normal(Vector3d::Zero()), m_curvature(0.0)
+    {}
+
+    Vector3d m_normal;
+    double m_curvature;
+
+    bool empty() const { return m_normal.isZero(); }
+
+    void calcNormal(PointView& view, PointIdList neighbors)
```

Comment:
The name of this function is weird. The class name is `NormalResult`. `calcNormal` seems redundant.